src/test/kotlin/com/atlassian/performance/tools/jvmtasks/api/TaskTimerTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.atlassian.performance.tools.jvmtasks.api
+
+import com.atlassian.performance.tools.jvmtasks.api.TaskTimer.time
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.time.Duration
+
+class TaskTimerTest {
+    private val handler = DummyEventHandler()
+
+    @Before
+    fun setUp() {
+        TaskTimer.registerTaskStartedHandler(handler)
+        TaskTimer.registerTaskSucceededHandler(handler)
+        TaskTimer.registerTaskFailedHandler(handler)
+    }
+
+    @After
+    fun tearDown() {
+        TaskTimer.deregisterTaskStartedHandler(handler)
+        TaskTimer.deregisterTaskSucceededHandler(handler)
+        TaskTimer.deregisterTaskFailedHandler(handler)
+    }
+
+    @Test
+    fun shouldReceiveEvent() {
+        time("Success") {
+        }
+
+        assertThat(handler.taskStartedCalls).containsOnly(listOf("Success"))
+        assertThat(handler.taskSucceededCalls).containsOnly(listOf("Success"))
+        assertThat(handler.taskFailedCalls).isEmpty()
+    }
+
+    @Test
+    fun shouldReceiveEventWhenTaskFails() {
+        val exception =
+            try {
+                time("Fail") {
+                    throw Exception("test exception")
+                }
+                @Suppress("UNREACHABLE_CODE")
+                null
+            } catch (e: Exception) {
+                e
+            }
```

Comment:
val exception = catchThrowable {
```suggestion
        val exception = catchThrowable {
            time("Fail") {
                throw Exception("test exception")
            }
        }
```
}